pkg/server/api/agent/v1/service.go

```diff
@@ -471,3 +618,206 @@ func (s *Service) getSelectorsFromAgentID(ctx context.Context, agentID string) (
 
 	return api.NodeSelectorsToProto(resp.Selectors)
 }
+
+func validateAttestAgentParams(params *agent.AttestAgentRequest_Params) error {
+	if params == nil {
+		return errors.New("missing params")
+	}
+	if params.Data == nil {
+		return errors.New("missing attestation data")
+	}
+	if params.Params == nil {
+		return errors.New("missing X509-SVID parameters")
+	}
+	if len(params.Params.Csr) == 0 {
+		return errors.New("missing CSR")
+	}
+	if params.Data.Type == "" {
+		return errors.New("missing attestation data type")
+	}
+	if params.Data.Payload == "" {
+		return errors.New("missing attestation data payload")
+	}
+	return nil
+}
+
+func (s *Service) isAttested(node *common.AttestedNode, baseSpiffeID string) bool {
```

Comment:
Is this method necessary? Seems like the node was just fetched with the spiffe id, so a simple `node != nil` check above seems ok?

Reply:
💯 I will remove and simply it.